UaClient/ServiceModel/Ua/Channels/UaTcpSessionChannel.cs

```diff
@@ -589,6 +589,11 @@ protected override async Task OnOpenAsync(CancellationToken token = default(Canc
                 }
 
                 var userNameIdentity = (UserNameIdentity)this.UserIdentity;
+                if (userNameIdentity.UserName == null && userNameIdentity.Password == null)
```

Comment:
Shouldn't that be ORed, i.e., `userNameIdentity.UserName == null || userNameIdentity.Password == null`?

Reply:
Wasn't sure. Is it possible to have authentication with a username and no password?

Reply:
Well I guess you would always want to check the password, or else you'll get the exception so yeah, I'll `or` it

Reply:
That was my thought, too.